path.c

```diff
@@ -1586,9 +1586,9 @@ char *xdg_config_home_for(const char *subdir, const char *filename)
 		home_config = mkpathdup("%s/.config/%s/%s", home, subdir, filename);
 
 	#ifdef WIN32
-	appdata = getenv("APPDATA");
+	const char *appdata = getenv("APPDATA");
```

Comment:
Please enclose this section within `{` and `}` and an extra indentation level to avoid the error about "ISO C90 forbids mixed declarations and code".

Reply:
I hope the commit message is more descriptive now. I tried to add all the context I could to justified this change.

Reply:
Yes, it is much better now, thank you!